File: Frontend/gvpp/src/models/Branch.ts
import type {Room} from "@/models/Room.ts";

export class Branch {
  id: number;
  name: string;
  description: string;
  address: string;
  postalCode: number;
  city: string;
  email: string;
  phoneNumber: string;
  website: string;
  rooms: Room[];


  constructor(
    id: number,
    name: string,
    description: string,
    address: string,
    postalCode: number,
    city: string,
    email: string,
    phoneNumber: string,
    website: string,
    rooms: Room[]
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.address = address;
    this.postalCode = postalCode;
    this.city = city;
    this.phoneNumber = phoneNumber;
    this.website = website;
    this.email = email;
    this.rooms = rooms;
  }
}

File: Backend/generic-free-spots-backend/src/main/java/be/PXLResearch/code4belgium/general/domain/Branch.java
package be.PXLResearch.code4belgium.general.domain;

import be.PXLResearch.code4belgium.enums.City;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public abstract class Branch<T, U> {
    @Id
    @GeneratedValue
    private Long id;

    private String name;
    private String description;

    private String address;
    private int postalCode;

    @Enumerated(EnumType.STRING)
    private City city;
    private String email;
    private String phoneNumber;
    private String website;
    @ManyToOne
    @JoinColumn(name = "organization_id")
    private T organization;

    @OneToMany(mappedBy = "branch")
    @JsonManagedReference
    private List<U> rooms;

}
